pkg/machinery/config/types/v1alpha1/v1alpha1_provider.go

```diff
@@ -440,13 +436,9 @@ func (n *NetworkConfig) GetDevice(name string) *Device {
 		}
 	}
 
-	dev := &Device{
+	return &Device{
 		DeviceInterface: name,
 	}
-
-	n.NetworkInterfaces = append(n.NetworkInterfaces, dev)
```

Comment:
```
// GetDevice adds or returns existing Device by name.
```

It no longer "adds" device. Not sure if some caller depends on it

Reply:
not sure either, good question, but certainly we need to stop doing that, I will do more review to see how it got actually used

Reply:
nice catch, this change was wrong, and this method is not used to access the config.